src/bin/eslint_js.sh

```
@@ -0,0 +1,161 @@
+    echo "    -d                   Fix all changed JS Files (Dry Run: just show violations)"
+}
+
+function sniff()
+{
+    if [ ! -z "$targetBranch" ]
+    then
+        parameter=" -t $targetBranch"
+    fi
+
+    changedJsFiles=$($findFilesCommand$parameter)
+
+    if [ -z "$changedJsFiles" ]
+    then
+        echo "No JS-Files to check!"
+        exit 0
+    fi
+
+    echo "########### ESLINT ###########"
+    find_violations
+
+    if [ "$findViolationsResult" -ne "0" ] && [ "$fixMode" == "true" ]
+    then
+        echo "########### ESLINT FIX ###########"
+        echo "Trying to fix violations"
+        fix_violations
+
+        if [ "$fixDryMode" == "false" ]
+        then
+            echo "Revalidating code after attempt to fix it"
+            find_violations
+
+            if [ "$findViolationsResult" -ne "0" ]
+            then
+                printf "\n\e[31mCouldn't fix all violations :( \e[0m\n"
+            else
+                printf "\n\e[32mFixed all Violations :D \e[0m\n"
+            fi
+        fi
+    fi
+
+    exit $findViolationsResult
+}
+
+function find_violations()
+{
+    findViolationsResult=""
+    local result=0
+
+    for directory in $changedJsFiles; do
+        echo "ESLinting $directory"
+        $sniffCommand $directory
+        sniffResult=$?
+        if ! [ "$sniffResult" -eq "0" ]
+        then
+            result=$sniffResult
+        fi
+    done
+
+    findViolationsResult="$result"
+}
+
+function fix_violations()
+{
+    local eslintCommand
+
+    if [ "$fixDryMode" == "true" ]
+    then
+        eslintCommand=$eslintFixDryCommand
+        echo "Because this is a dry run violations will not really be fixed!"
+    else
+        eslintCommand=$eslintFixCommand
+    fi
+
+    for directory in $changedJsFiles; do
+        echo "Fixing $directory"
+        $eslintCommand $directory
+        sniffResult=$?
+    done
+}
+
+###############
+
+constructor
+
+while getopts "hfdt:" opt; do
```

Doppelpunkt for dem Options-String aktiviert "silent error reporting":

```while getopts ":hfdt:" opt; do
```